Submitty/Submitty
# [Feature: Forum] Create Socket Server script and Push threads/posts to users

### Please check if the PR fulfills these requirements:


### What is the current behavior?
Partially Closes #4328 

### What is the new behavior?
Addition of Web Socket to the system, which is connected to by the discussion Forum threads listing page. 

### How to test: 

1. `vagrant ssh`
2. `php site/app/controllers/SocketController.php`
3. Open the Discussion Forum
4. Enable the Chrome Console
5. Open another instance of Forum on another browser/incognito
6. Make replies to posts, create threads to observe Forum updates
